main/src/library/RedBlackTree.flix

```diff
@@ -121,10 +121,12 @@ namespace RedBlackTree {
     /// Returns the number of nodes in `t`.
     ///
     @Time(size(t)) @Space(Int32.log2(size(t)))
-    pub def size(t: RedBlackTree[k, v]): Int32 = match t {
```

Comment:
Technically, this does not have to be tail recursive because the tree has height O(log n) which means the stack can never be blown.